lib/Core/Search/Common/EventSubscriber/ContentEventSubscriber.php

```diff
@@ -55,26 +59,24 @@ public function onCopyContent(CopyContentEvent $event): void
 
     public function onBeforeDeleteContent(BeforeDeleteContentEvent $event): void
     {
+        $contentLocations = $this->locationService->loadLocations($event->getContentInfo());
```

Comment:
Using `LocationService` here means it will use current user's permissions, and we don't want it to use permissions at all. Please use Location handler from Persistence instead.

Reply:
resolved in https://github.com/netgen/ibexa-search-extra/pull/9/commits/0d993e7b239d7830b45ae363fc2bfd8f049fd764

Reply:
Compiler pass also needs to be updated for the change in the signature. Also, in persistence Location parent ID property is `parentId`.

Also please inject Location handler directly, it's a good practice to do inject exactly what is needed.

Reply:
fixed in 7e34ad4302ec53504ae345759b7d4ee6b3e8ae85